### PR TITLE
Community Branding Implementation

### PR DESCRIPTION
**Pull request by PaRaDoX50**
Implemented Basic and Advanced branding using Local DB (Hive), Models, and Entities.
Used a mock repository for branding data because it is a part of the data SDK.

### CODE REVIEW

File: lib/src/data/local_db_service.dart [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import 'package:feed_sx/src/models/advanced_branding.dart';
```
Comment (divyanshgandhilm) [inline lib/src/data/local_db_service.dart:1]: If this is a service, it should be in the services folder. The repository can be here.

File: lib/src/sdk/sdk_service.dart [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// ignore_for_file: public_member_api_docs, sort_constructors_first
```
Comment (divyanshgandhilm) [inline lib/src/sdk/sdk_service.dart:1]: Why is this file called SDK service? Also let's format the files properly with proper return statements and spacing.

File: lib/src/theme.dart [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+const double kButtonFontSize = 12;
+const double kFontXSmall = 11;
+const double kFontMedium = 16;
+const double kPaddingXSmall = 2;
+const double kPaddingSmall = 4;
+const double kPaddingMedium = 8;
+const double kPaddingLarge = 16;
+const double kPaddingXLarge = 20;
+const double kBorderRadiusXSmall = 2;
+const SizedBox kHorizontalXLargeP = SizedBox(width: kPaddingXLarge);
```
Comment (divyanshgandhilm) [inline lib/src/theme.dart:20]: Really nice addition, this will make it easier!

**Review by divyanshgandhilm — CHANGES_REQUESTED** [review]
A few changes, and comments

**Review by divyanshgandhilm — APPROVED** [review]

**Comment by divyanshgandhilm** [comment]
Let's let @NateshR take a final look as well before merging

File: lib/src/data/local/local_db_impl.dart [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    return branding;
+  }
+
+  @override
+  Future<void> saveBranding({required BrandingEntity branding}) async {
+    var brandingBox = await Hive.openBox(brandingBoxKey);
+    if (brandingBox.containsKey(brandingKey)) {
+      await brandingBox.delete(brandingKey);
```
Comment (NateshR) [inline lib/src/data/local/local_db_impl.dart:41]: don't we have an update method?

File: lib/src/data/local/local_db_impl.dart [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import 'package:feed_sx/src/data/local/local_db.dart';
+import 'package:feed_sx/src/data/models/advanced_branding.dart';
+import 'package:feed_sx/src/data/models/basic_branding.dart';
+import 'package:feed_sx/src/data/models/branding.dart';
+
+import 'package:hive_flutter/hive_flutter.dart';
+
+class LocalDBImpl extends LocalDB {
+  static const getItInstanceName = "local_db_service";
+  final String brandingBoxKey = 'branding_box';
```
Comment (NateshR) [inline lib/src/data/local/local_db_impl.dart:10]: two keys have same value?

File: lib/src/data/models/advanced_branding.dart [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+  }
+  BrandingAdvancedEntity toEntity() {
+    return BrandingAdvancedEntity(
+        headerColor: headerColor?.value.toString(),
+        buttonIconsColor: buttonIconsColor?.value.toString(),
+        textLinksColor: textLinksColor?.value.toString());
+  }
+
+  @override
+  String toString() =>
```
Comment (NateshR) [inline lib/src/data/models/advanced_branding.dart:34]: what's the use of this

File: lib/src/theme.dart [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
```
Comment (NateshR) [inline lib/src/theme.dart:20]: sync name

**Review by NateshR — CHANGES_REQUESTED** [review]

**Comment by PaRaDoX50** [comment]
@NateshR Made the desired changes. PTAL.

**Review by NateshR — APPROVED** [review]

**Comment by divyanshgandhilm** [comment]
Everything looks good. One thing, why did we create a src folder inside the lib folder? The lib folder does the same job as you are doing right now with src right?

**Review by divyanshgandhilm — APPROVED** [review]

**Pull request merged**